openedx/core/djangoapps/user_api/preferences/tests/test_views.py

```diff
@@ -0,0 +1,392 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for preference APIs.
+"""
+
+import unittest
+import ddt
+import json
+
+from django.core.urlresolvers import reverse
+from django.conf import settings
+
+from openedx.core.djangoapps.user_api.models import UserPreference
+from openedx.core.djangoapps.user_api.accounts.tests.test_views import UserAPITestCase
+
+TOO_LONG_PREFERENCE_KEY = "x" * 256
+
+
+@ddt.ddt
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class TestPreferencesAPI(UserAPITestCase):
+    """
+    Unit tests /api/user/v0/accounts/{username}/
+    """
+    def setUp(self):
+        super(TestPreferencesAPI, self).setUp()
+        self.url_endpoint_name = "preferences_api"
+        self.url = reverse(self.url_endpoint_name, kwargs={'username': self.user.username})
+
+    def test_anonymous_access(self):
```

Comment:
Add a test points for unsupported methods. For example, this is the accounts test:

    def test_unsupported_methods(self):
        """
        Test that DELETE, POST, and PUT are not supported.
        """
        self.client.login(username=self.user.username, password=self.test_password)
        self.assertEqual(405, self.client.put(self.url).status_code)
        self.assertEqual(405, self.client.post(self.url).status_code)
        self.assertEqual(405, self.client.delete(self.url).status_code)